pkgs/by-name/co/cook-cli/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildNpmPackage
+, rustPlatform
+, pkg-config
+, openssl
+, darwin
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cook-cli";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "cooklang";
+    repo = "cookcli";
+    rev = "v${version}";
+    hash = "sha256-3gLVsk6GCxOG24Md7E9fk28Vnc4kVDdwyZUD/GtSwFE=";
+  };
+
+  cargoHash = "sha256-6lnURuE1cgNAniHl5ozXo1W3cLYYje7er+ZhvZDKdVg=";
+
+  nativeBuildInputs = [ pkg-config openssl ];
+
+  buildInputs =
+    [ openssl ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
+
+  postPatch = ''
+    cp -r ${ui}/* "ui/public"
+  '';
+
+  OPENSSL_NO_VENDOR = 1;
+
+  ui = buildNpmPackage {
```

Comment:
instead of doing ui like this it would be cool if you could take a look at how we do it in [alice-lg](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/servers/alice-lg/default.nix#L26) using `passthru` because that later allow the user to overwrite the frontend separately

Reply:
oh cool! Didn't know about `passthru` and overrides. Makes sense.